frontend/src/service-worker.js

```diff
@@ -1,15 +1,91 @@
-import {serviceWorkerFetchListener} from "sync-message";
+/* eslint-disable */
+// Otherwise webpack can fail silently
+// https://github.com/facebook/create-react-app/issues/8014
+// Reference for service-worker.js: https://developers.google.com/web/tools/workbox/modules
 
-console.log(self.__WB_MANIFEST);
+import {serviceWorkerFetchListener} from 'sync-message'; // Do not remove
+const fetchListener = serviceWorkerFetchListener(); // Do not remove
+addEventListener('fetch', fetchListener); // Do not remove
 
-const fetchListener = serviceWorkerFetchListener();
+import {clientsClaim} from 'workbox-core';
+import {ExpirationPlugin} from 'workbox-expiration';
+import {precacheAndRoute, createHandlerBoundToURL} from 'workbox-precaching';
+import {registerRoute} from 'workbox-routing';
+import {NetworkFirst, StaleWhileRevalidate} from 'workbox-strategies';
 
-addEventListener('fetch', fetchListener);
+clientsClaim();
 
-addEventListener('install', function (e) {
-  e.waitUntil(self.skipWaiting());
-});
+if (process.env.NODE_ENV === 'production') {
+  // Precache all of the assets generated by your build process.
+  // Their URLs are injected into the manifest variable below.
+  // This variable must be present somewhere in your service worker file,
+  // even if you decide not to use precaching. See https://cra.link/PWA
+  precacheAndRoute(self.__WB_MANIFEST);
+}
+
+// Set up App Shell-style routing, so that all navigation requests
+// are fulfilled with your index.html shell. Learn more at
+// https://developers.google.com/web/fundamentals/architecture/app-shell
+const fileExtensionRegexp = new RegExp('/[^/?]+\\.[^/]+$');
+registerRoute(
+  // Return false to exempt requests from being fulfilled by index.html.
+  ({request, url}) => {
+    // If this isn't a navigation, skip.
+    if (request.mode !== 'navigate') {
+      return false;
+    }
+
+    // If this is a URL that starts with /_, skip.
+    if (url.pathname.startsWith('/_')) {
+      return false;
+    } // If this looks like a URL for a resource, because it contains // a file extension, skip.
+
+    if (url.pathname.match(fileExtensionRegexp)) {
+      return false;
+    }
+
+    // Return true to signal that we want to use the handler.
+    return true;
+  },
+  createHandlerBoundToURL(process.env.PUBLIC_URL + '/index.html'),
+);
+
+if (process.env.NODE_ENV === 'production') {
+  registerRoute(
+    ({url}) => {
+      const urlString = url.toString();
+      if (
+        urlString.startsWith('https://cdn.jsdelivr.net/') // Pyodide
+        || urlString.startsWith('https://pyodide-cdn2.iodide.io') // Only used when we are testing bleeding-edge pyodide
+        || urlString.startsWith('https://cdnjs.cloudflare.com') // Only used by homepage, so probably not needed here
+        || urlString.startsWith('https://fonts.googleapis.com') // Only used by homepage, so probably not needed here
```

Comment:
@alexmojaki what do you think, shall I remove these?

```
        || urlString.startsWith('https://cdnjs.cloudflare.com') // Only used by homepage, so probably not needed here
        || urlString.startsWith('https://fonts.googleapis.com') // Only used by homepage, so probably not needed here
```

Reply:
Yeah, remove them.